controllers/array_action/array_mediator_svc.py

```diff
@@ -2064,7 +2068,58 @@ def remove_volume_from_volume_group(self, volume_id):
         cli_volume = self._get_cli_volume_by_wwn(volume_id, not_exist_err=True)
         self._change_volume_group(cli_volume.id, None)
 
-    def register_plugin(self, unique_key,  metadata):
+    def _get_ownership_group_pools(self, ownership_group):
+        logger.info(svc_messages.GET_OWNERSHIP_GROUP_POOLS.format(ownership_group))
+        filter_value = 'owner_name={}'.format(ownership_group)
+        cli_pools = self._lsmdiskgrp(filtervalue=filter_value).as_list
+        return cli_pools
+
+    def is_fenced(self, fence_ownership_group):
+        ownership_group_pools = self._get_ownership_group_pools(fence_ownership_group)
+        if len(ownership_group_pools) == 0:
+            logger.info(svc_messages.NO_POOLS_FOUND_IN_OWNERSHIP_GROUP.format(fence_ownership_group))
+            return True
+
+        logger.info(svc_messages.POOLS_FOUND_IN_OWNERSHIP_GROUP.format(fence_ownership_group, ownership_group_pools))
+        return False
+
+    def _chmdiskgrp(self, pool_id, **cli_kwargs):
+        self.client.svctask.chmdiskgrp(object_id=pool_id, **cli_kwargs)
+
+    def _remove_all_mappings_from_ownership_group(self, ownership_group):
+        logger.info(svc_messages.REMOVING_ALL_MAPPINGS_FROM_OWNERSHIP_GROUP.format(ownership_group))
+        filter_value = 'owner_name={}'.format(ownership_group)
+
+        hosts = self.client.svcinfo.lshost(filtervalue=filter_value).as_list
+        host_names = [host.name for host in hosts]
+
+        volumes = self._lsvdisk_list(filtervalue=filter_value)
+        volume_names = [volume.name for volume in volumes]
+
+        mappings = self.client.svcinfo.lshostvdiskmap().as_list
+
+        relevant_mappings = [mapping for mapping in mappings if
+                             mapping.name in host_names and mapping.vdisk_name in volume_names]
+        logger.info(svc_messages.REMOVING_MAPPINGS.format(relevant_mappings))
+        for mapping in relevant_mappings:
+            self.client.svctask.rmvdiskhostmap(vdisk_name=mapping.vdisk_name, host=mapping.name)
```

Comment:
you can export this code into some functions, it can make it more cleaner IMO, but it's not important, it's not a blocker for the PR